workitem_blackbox_test.go

```diff
@@ -89,6 +91,39 @@ func TestGetWorkItemWithLegacyDescription(t *testing.T) {
 	test.DeleteWorkitemOK(t, nil, nil, controller, *result.Data.ID)
 }
 
+func TestReorderWorkItem(t *testing.T) {
+	resource.Require(t, resource.Database)
+	pub, _ := almtoken.ParsePublicKey([]byte(almtoken.RSAPublicKey))
+	priv, _ := almtoken.ParsePrivateKey([]byte(almtoken.RSAPrivateKey))
+	svc := testsupport.ServiceAsUser("TestGetWorkItem-Service", almtoken.NewManager(pub, priv), testsupport.TestIdentity)
+	assert.NotNil(t, svc)
+	controller := NewWorkitemController(svc, gormapplication.NewGormDB(DB))
+	assert.NotNil(t, controller)
+	payload1 := minimumRequiredCreateWithType(workitem.SystemBug)
+	payload1.Data.Attributes[workitem.SystemTitle] = "Reorder Test WI"
+	payload1.Data.Attributes[workitem.SystemState] = workitem.SystemStateClosed
+
+	_, result1 := test.CreateWorkitemCreated(t, svc.Context, svc, controller, &payload1)
+	_, result2 := test.CreateWorkitemCreated(t, svc.Context, svc, controller, &payload1)
+	_, result3 := test.CreateWorkitemCreated(t, svc.Context, svc, controller, &payload1)
+	payload2 := minimumRequiredReorderPayload()
+	var dataArray []*app.WorkItem2
+	dataArray = append(dataArray, result2.Data, result3.Data)
+	payload2.Data = dataArray
+	payload2.Position.Above = *result1.Data.ID
+
+	_, reordered1 := test.ReorderWorkitemOK(t, nil, nil, controller, &payload2)
+	assert.NotNil(t, reordered1.Data)
+	assert.Equal(t, result2.Data.Attributes["version"].(int)+1, reordered1.Data[0].Attributes["version"])
+	assert.Equal(t, result3.Data.Attributes["version"].(int)+1, reordered1.Data[1].Attributes["version"])
+	assert.Equal(t, *result2.Data.ID, *reordered1.Data[0].ID)
+	assert.Equal(t, *result3.Data.ID, *reordered1.Data[1].ID)
+
+	test.DeleteWorkitemOK(t, nil, nil, controller, *result1.Data.ID)
```

Comment:
If these three deletes are supposed to cleanup all created work items, then it would be a good idea to have these lines in a defer statement right after they have been created. We have different patterns for cleaning up in all of our tests and while this may not be the nicest one it is still better than "leaking" entries in the DB.